test/sequential/test-fs-watchfile.js

```diff
@@ -64,23 +64,47 @@ fs.watchFile(enoentFile, {interval: 0}, common.mustCall(function(curr, prev) {
   }
 }, 2));
 
-// Watch events should callback with a filename on supported systems
-if (common.isLinux || common.isOSX || common.isWindows || common.isAix) {
-  const dir = common.tmpDir + '/watch';
 
-  fs.mkdir(dir, common.mustCall(function(err) {
-    if (err) assert.fail(err);
-
-    fs.watch(dir, common.mustCall(function(eventType, filename) {
-      clearInterval(interval);
-      this._handle.close();
-      assert.strictEqual(filename, 'foo.txt');
-    }));
+function WatchTestCase(is, dirName, fileName, field) {
+  Object.defineProperties(this, {
+    dirPath: { get: () => `${common.tmpDir}/${this.dirName}` },
+    filePath: { get: () => `${this.dirPath}/${this.fileName}` }
+  });
```

Comment:
Any reason to not turn this into an ES6 class and use fancy `get dirPath() { … }` syntax? :)